src/main/java/games/strategy/triplea/util/BonusIncomeUtils.java

```diff
@@ -21,13 +22,25 @@ public static String addBonusIncome(final IntegerMap<Resource> income, final IDe
     for (final Resource resource : income.keySet()) {
       final int amount = income.getInt(resource);
       final int incomePercent = Properties.getIncomePercentage(player, bridge.getData());
-      final int bonusIncome = (int) Math.round(((double) amount * (double) (incomePercent - 100) / 100));
+      int puIncomeBonus = 0;
+      if (resource.getName().equals(Constants.PUS)) {
+        puIncomeBonus = Properties.getPuIncomeBonus(player, bridge.getData());
+      }
+      int bonusIncome = (int) Math.round(((double) amount * (double) (incomePercent - 100) / 100)) + puIncomeBonus;
       if (bonusIncome == 0) {
         continue;
       }
       final int total = player.getResources().getQuantity(resource) + bonusIncome;
-      final String message = "Giving " + player.getName() + " " + (incomePercent - 100) + "% bonus income of "
-          + bonusIncome + " " + resource.getName() + "; end with " + total + " " + resource.getName();
+      String message = "Giving " + player.getName();
+      if (puIncomeBonus > 0) {
+        message += " " + puIncomeBonus + " &";
```

Comment:
Should this message include a `resource.getName()` after the value?  It was a little confusing when I viewed the initialization delegate messages in the history for a game I tested:

> Giving Russians 50 & 150% income for 62 PUs; end with 86 PUs

I think it could be improved with units:

> Giving Russians 50 PUs & 150% income for 62 PUs; end with 86 PUs